gamestonk_terminal/cryptocurrency/coingecko/pycoingecko_view.py

```diff
@@ -0,0 +1,1570 @@
+        - Most voted coins
+        - Most popular coins
+        - Recently added coins
+        - Most positive sentiment coins
+
+    Parameters
+    ----------
+    category: str
+        one from list: [trending, most_voted, positive_sentiment, most_visited]
+    other_args: List[str]
+        Arguments to pass to argparse
+
+    """
+    parser = argparse.ArgumentParser(
+        prog="discover_coins",
+        add_help=False,
+        description=f"Discover {category} coins by one of category\n"
+        "Available categories: [trending, most_voted, positive_sentiment, most_visited]",
+    )
+
+    try:
+        ns_parser = parse_known_args_and_warn(parser, other_args)
+        if not ns_parser:
+            return
+
+        df = gecko.discover_coins(category=category)
+        df.index = df.index + 1
+        df.reset_index(inplace=True)
+        df.rename(columns={"index": "rank"}, inplace=True)
+        print(
+            tabulate(
+                df,
+                headers=df.columns,
+                floatfmt=".5f",
+                showindex=False,
+                tablefmt="fancy_grid",
+            )
+        )
+        print("")
+
+    except Exception as e:
+        print(e)
+        print("")
+
+
+def news(other_args: List[str]):
+    """Shows latest crypto news from www.coingecko.com
+
+    Parameters
+    ----------
+    other_args: List[str]
+        Arguments to pass to argparse
+
+    """
+    parser = argparse.ArgumentParser(
+        prog="news",
+        add_help=False,
+        description="Shows latest crypto news from CoinGecko",
+    )
+
+    parser.add_argument(
+        "-t",
+        "--top",
+        dest="top",
+        type=int,
+        help="top N number of news >=10",
+        default=100,
+    )
+
+    try:
+        ns_parser = parse_known_args_and_warn(parser, other_args)
+        if not ns_parser:
+            return
+
+        df = gecko.get_news(n=ns_parser.top)
+        print(
+            tabulate(
+                df,
+                headers=df.columns,
+                floatfmt=".0f",
+                showindex=False,
+                tablefmt="fancy_grid",
+            )
+        )
+        print("")
+
+    except Exception as e:
+        print(e)
+        print("")
+
+
+def top_categories(other_args: List[str]):
+    """Shows top cryptocurrency categories by market capitalization from https://www.coingecko.com/en/categories
+    The cryptocurrency category ranking is based on market capitalization.
+
+    Parameters
+    ----------
+    other_args: List[str]
+        Arguments to pass to argparse
+
+    """
+    parser = argparse.ArgumentParser(
+        prog="top_categories",
+        add_help=False,
+        description="Shows top cryptocurrency categories by market capitalization",
+    )
+
+    try:
+        ns_parser = parse_known_args_and_warn(parser, other_args)
+        if not ns_parser:
+            return
+
+        df = gecko.get_top_crypto_categories()
```

Comment:
One idea for these tables is to try to shorten the zeros of the billions. You can see how it's done in the fundamental analysis menu.

Instead of displaying `1,123,123,123 $`. We do `1.123 B$`